jobs/jobs.go

```diff
@@ -286,16 +288,18 @@ func (job *Job) processEvent(ctx context.Context, event events.Event) bool {
 			break
 		}
 		log.Debugf("job exited but restart not permitted: %v", job.Name)
+		job.startEvent = events.NonEvent
 		return true
 	case job.startEvent:
 		if job.startsRemain == 0 {
+			job.startEvent = events.NonEvent
```

Comment:
We need it for the `return true`; the return value gets used by the caller to determine whether to exit the job entirely and stop processing events.

Adding the `job.startEvent = events.NonEvent` assignment in this block isn't strictly necessary here as a result, but being I want to be consistent about how `processEvent` changes the state of the `Job` without concern for the callers (if for no other reason than it makes for easier testing).